### PR TITLE
ch(precommitscript): Configure ESLint and pre-commit script

### PR DESCRIPTION
**Pull request by Francoise02**
What does this PR do?
It checks the style violations and flag them.

How should this be manually tested?
Clone this repo, write codes, then hover around the codes. If there's any violation in style ESlint will highlight it. 

Any background context you want to provide?
This ESlint does not consider the typescript

what are the relevant pivotal tracker stories?
[#182201510](https://www.pivotaltracker.com/n/projects/2569104/stories/182201510)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+//     },
+//     "plugins": [
+//         "react"
+//     ],
+//     "rules": {
+//     }
+// }
+
+{
+
```
Comment (didasmbalanya) [inline .eslintrc.json]: remove commented code

**Review by didasmbalanya — CHANGES_REQUESTED** [review]
Missing pre commit hook which should format the code and run test

File: .eslintrc.json (hunk 1 above)
Comment (Francoise02) [reply .eslintrc.json]: Sure. I have done git rebase and will remove those commented codes soon. However, the pre-commit that will run the tests is blocked by the setup testing but I'll add it as soon as the setup testing is merged.

**Review by ishimweeli — DISMISSED** [review]

**Comment by Francoise02** [comment]
All ESLint, Prettier, and the pre-commit hook for code formatting function properly. Although, the pre-commit hook for running the tests will be completed after the setup-testing branch is merged to dev.

**Review by Julienish14 — DISMISSED** [review]
great

File: pull_request_template.md [hunk 2]
```diff
@@ -18,3 +19,14 @@
...
+https://www.pivotaltracker.com/n/projects/2569104/stories/182201510
+=======
+This branch checks code formating violations using ESlint & formats the code by running prettier formatter before pushing any changes to the repo
+
+To test it you can easily clore the repo --> go to the ch-precommitscript-182201510 branch --> Make any changes in any files --> Hover around any the codes, if there are any code format violations you will see them highlighted by ESlint --> After checking those, you can now commit to the repo by running git commit and the codes will all be formatted automatically by prettier.
+
+Please note that you have to install Eslint and prettier extensions on your IDE for the code formatting to run effectively
+
+Here is the relevant pivotal tracker stories: https://www.pivotaltracker.com/n/projects/2569104/stories/182201510
+>>>>>>> 905131f (ch(precommitscript): Configure ESLint and pre-commit script)
```
Comment (didasmbalanya) [inline pull_request_template.md]: This is not supposed to be changed

**Review by didasmbalanya — CHANGES_REQUESTED** [review]

**Review by Julienish14 — DISMISSED** [review]

**Review by kalisaNkevin — APPROVED** [review]

**Review by munyanezaarmel — APPROVED** [review]

**Comment by munyanezaarmel** [comment]
Well done @Francoise02

**Review by didasmbalanya — APPROVED** [review]

**Pull request merged**